bin/reth/src/chain/import.rs

```diff
@@ -0,0 +1,168 @@
+
+        // create a new FileClient
+        info!(target: "reth::cli", "Importing chain file");
+        let file_client = Arc::new(FileClient::new(&self.path).await?);
+
+        // override the tip
+        let tip = file_client.tip().expect("file client has no tip");
+        info!(target: "reth::cli", "Chain file imported");
+
+        let (consensus, notifier) = BeaconConsensus::builder().build(self.chain.clone());
+        debug!(target: "reth::cli", %tip, "Tip manually set");
+        notifier.send(ForkchoiceState {
+            head_block_hash: tip,
+            safe_block_hash: tip,
+            finalized_block_hash: tip,
+        })?;
+        info!(target: "reth::cli", "Consensus engine initialized");
+
+        let (mut pipeline, events) =
+            self.build_import_pipeline(config, db.clone(), &consensus, file_client).await?;
+
+        tokio::spawn(handle_events(events));
+
+        // Run pipeline
+        info!(target: "reth::cli", "Starting sync pipeline");
+        tokio::select! {
+            res = pipeline.run(db.clone()) => res?,
+            _ = tokio::signal::ctrl_c() => {},
+        };
+
+        info!(target: "reth::cli", "Finishing up");
+        Ok(())
+    }
+
+    async fn build_import_pipeline<C>(
+        &self,
+        config: Config,
+        db: Arc<Env<WriteMap>>,
+        consensus: &Arc<C>,
+        file_client: Arc<FileClient>,
+    ) -> eyre::Result<(Pipeline<Env<WriteMap>, impl SyncStateUpdater>, impl Stream<Item = NodeEvent>)>
+    where
+        C: Consensus + 'static,
+    {
+        let header_downloader = ReverseHeadersDownloaderBuilder::from(config.stages.headers)
+            .build(consensus.clone(), file_client.clone())
+            .as_task();
+
+        let body_downloader = BodiesDownloaderBuilder::from(config.stages.bodies)
+            .build(file_client.clone(), consensus.clone(), db)
+            .as_task();
+
+        let mut pipeline = Pipeline::builder()
+            .with_sync_state_updater(file_client)
+            .add_stages(
+                OnlineStages::new(consensus.clone(), header_downloader, body_downloader).set(
+                    TotalDifficultyStage {
+                        chain_spec: self.chain.clone(),
+                        commit_threshold: config.stages.total_difficulty.commit_threshold,
+                    },
+                ),
+            )
+            .add_stages(
+                OfflineStages::default()
+                    .set(SenderRecoveryStage {
+                        batch_size: config.stages.sender_recovery.batch_size,
+                        commit_threshold: config.stages.sender_recovery.commit_threshold,
+                    })
+                    .set(ExecutionStage {
+                        chain_spec: self.chain.clone(),
+                        commit_threshold: config.stages.execution.commit_threshold,
+                    }),
+            )
+            .with_max_block(0)
```

Comment:
Is this right? Don't we want to run the pipeline too? Will this run the headers and bodies stages as is?

Reply:
We want to run the pipeline

Reply:
yeah, that would be valuable